lib/bundler/rubygems_integration.rb

```diff
@@ -461,8 +461,8 @@ def backport_spec_file
       end
     end
 
-    def redefine_method(klass, method, &block)
-      if klass.instance_methods(false).include?(method)
+    def redefine_method(klass, method, force_removal = false, &block)
+      if force_removal || klass.instance_methods.include?(method)
```

Comment:
how exactly does this logic work or how is it intended to work? Currently, if `force_removal` is `true`, then it'll try to invoke `remove_method` regardless of whether that method is defined or not?